app/services/reports/region_cache_warmer.rb

```diff
@@ -1,69 +1,36 @@
 module Reports
   class RegionCacheWarmer
     prepend SentryHandler
+    BATCH_SIZE = 100
 
-    def self.call
-      new.call
+    def self.call(*args)
+      new(*args).call
     end
 
-    attr_reader :name
-    attr_reader :period
+    attr_reader :batch_size
 
-    def initialize(period: Reports.default_period)
-      @period = period
-      @name = self.class.name.to_s
-      notify "Starting #{name} warming"
+    def initialize(batch_size: BATCH_SIZE)
+      @batch_size = batch_size
     end
 
     def call
-      if Flipper.enabled?(:disable_region_cache_warmer)
-        notify "disabled via flipper - exiting"
-        return
-      end
```

Comment:
Let's keep the flipper check here too. Having it here as well as inside the job ensure that
* We're not wasteful with queuing and no-opping a bunch of jobs
* We can kill queued jobs in progress if needed halfway through a warm sesh

Reply:
> We can kill queued jobs in progress if needed halfway through a warm sesh

Unsure about this, how would this work?

Reply:
I mean - if we keep the feature flag inside the job also, then we can disable the flag, say, midway through a warming session, and the rest of the jobs in the queue would no-op. I don't know, in case DB CPU was spiking or something.

Reply:
Oh right. +1 for keeping in both places.